src/tracing/esp32_trace/esp32_tracing.cpp

```diff
@@ -157,32 +154,58 @@ void ESP32Backend::TraceCounter(const char * label)
 
 void ESP32Backend::LogMetricEvent(const MetricEvent & event)
 {
+    using ValueType = MetricEvent::Value::Type;
     if (!mRegistered)
```

Comment:
you have a single registered however several registration types. Is this ok?

i.e. if I log metrics like this:
   LogMetric(Int32)
   LogMetric(ErrorCode)

this will only ever register a "DATA_TYPE_INT" however then below it will send values of both int ant uint type.

"event.key" also varies throughout metrics ... we register only once for the first key we see, but then we try to upload data with every key. How does this work given that `mRegistered` is a single boolean value? Sounds like you would need some `set<tuple<string, type>>` to define what was registered and what not ... and that sounds potentially unbounded.

How is this supposed to work?

Reply:
Addressed in the latest change.